File: DAY-53/Remove Duplicate.cpp
#include<bits/stdc++.h>
using namespace std;

/*You are required to complete this function */

int remove_duplicate(int [],int );

int main()
{
    int T;
    cin>>T;
    while(T--)
    {
        int N;
        cin>>N;
        int a[N];
        for(int i=0;i<N;i++)
        {
            cin>>a[i];
        }

    int n = remove_duplicate(a,sizeof(a)/sizeof(a[0]));

    for(int i=0;i<n;i++)
        cout<<a[i]<<" ";
    cout<<endl;
    }
}
// } Driver Code Ends
/*You are required to complete this function */
int remove_duplicate(int A[],int N)
{
//Your code here
int cmp = A[0];
int i = 1;
for(int k = 1; k < N ; k++)
{
if(cmp != A[k])
{
A[i] = A[k];
cmp = A[i];
i++;
}
}
return i;
}

File: DAY-53/Remove duplicate.py
def remove_duplicate(n, arr):
    m=[]
    for i in range(n):
        m.append(arr[i])
    
    arr.clear()
    for i in m:
        if i not in arr:
            arr.append(i)
    return(len(arr))
    
#  Driver Code Starts
#Your code goes here
if __name__=='__main__':
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        n = remove_duplicate(n, arr)
        for i in range(n):
            print (arr[i], end=" ")
        print()


# } Driver Code Ends